src/transformers/modelcard.py

```diff
@@ -218,3 +230,277 @@ def to_json_file(self, json_file_path):
         """Save this instance to a json file."""
         with open(json_file_path, "w", encoding="utf-8") as writer:
             writer.write(self.to_json_string())
+
+
+AUTOGENERATED_COMMENT = """
+<!-- This model card has been generated automatically according to the information the Trainer had access to. You
+should probably proofread and complete it, then remove this comment. -->
+"""
+
+
+@dataclass
+class TrainingSummary:
+    model_name: str
+    language: Optional[str] = None
+    license: Optional[str] = None
+    pipeline_tag: Optional[str] = None
+    finetuned_from: Optional[str] = None
+    dataset: Optional[Union[str, List[str]]] = None
+    dataset_tags: Optional[Union[str, List[str]]] = None
+    eval_results: Optional[Dict[str, float]] = None
+    eval_lines: Optional[List[str]] = None
+    hyperparameters: Optional[Dict[str, Any]] = None
+
+    def to_model_card(self):
+        model_card = ""
+        metadata = ""
+        if self.language is not None:
+            metadata += f"language: {self.language}\n"
+        if self.license is not None:
+            metadata += f"license: {self.license}\n"
+        if self.pipeline_tag is not None:
+            metadata += f"pipeline-tag: {self.pipeline_tag}\n"
+        if self.dataset_tags is not None:
+            datasets = [self.dataset_tags] if isinstance(self.dataset_tags, str) else self.dataset_tags
+            datasets = [f"- {ds}" for ds in datasets]
+            metadata += "datasets:\n" + "\n".join(datasets)
+        if len(metadata) > 0:
+            model_card = f"---\n{metadata}\n---\n"
+
+        model_card += AUTOGENERATED_COMMENT
+
+        model_card += f"\n# {self.model_name}\n\n"
+
+        if self.finetuned_from is None:
+            model_card += "This model was trained from scratch on "
+        else:
+            model_card += f"This model is a fine-tuned version of [{self.finetuned_from}](https://huggingface.co/{self.finetuned_from}) on "
+
+        if self.dataset is None:
+            model_card += "an unkown dataset."
+        else:
+            if isinstance(self.dataset, str):
+                model_card += f"the {self.dataset} dataset."
+            else:
+                model_card += (
+                    ", ".join([f"the {ds}" for ds in self.dataset[:-1]]) + f" and the {self.dataset[-1]} datasets."
+                )
+
+        if self.eval_results is not None:
+            model_card += "\nIt achieves the following results on the evaluation set:\n"
+            model_card += "\n".join([f"- {name}: {_maybe_round(value)}" for name, value in self.eval_results.items()])
+            model_card += "\n"
+
+        if self.hyperparameters is not None:
+            model_card += (
+                "\n ## Training hyperparameters\n\nThe following hyperparameters were used during training:\n"
```

Comment:
```suggestion
                "\n## Training hyperparameters\n\nThe following hyperparameters were used during training:\n"
```

Reply:
In the auto-generated README.md I think all subsections starting with `##` were shifted one to the right